lib/medtronic600/medtronic600Simulator.js

```diff
@@ -493,6 +359,12 @@ class Medtronic600Simulator {
             return false;
           }
         }
+      } else if (event.type === 'basal') {
+        // Filter out any basals with no duration. This can happen inside
+        // applySuspendedBasals(), when a basal is suppressed completely by a pump suspend
```

Comment:
Yes, that's right. I did originally start putting the code in `applySuspendedBasals()`, however when I put `_.remove(this.events, event => event === activeSuspendedBasal);` in `applySuspendedBasals()`, eslint complained about [functions in loops](being bad). Perhaps we feel that it's for the best - if you agree, I can move it there.